e2e/cypress/integration/multi_team_and_dm/join_open_team_from_dm_spec.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Stage: @prod
+// Group: @multi_team_and_dm
+
+describe('Join an open team from a direct message link', () => {
+    let openTeam;
+    let testUserInOpenTeam;
+
+    let secondTestTeam;
+    let testUserOutsideOpenTeam;
+
+    before(() => {
+        cy.apiCreateTeam('mmt452-second-team', 'mmt452-second-team', 'I', true).then(({team}) => {
+            secondTestTeam = team;
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserOutsideOpenTeam = user;
+                cy.apiAddUserToTeam(secondTestTeam.id, testUserOutsideOpenTeam.id);
+            });
+        });
+
+        cy.apiCreateTeam('mmt452-open-team', 'mmt452-open-team', 'O', true).then(({team}) => {
+            openTeam = team;
+
+            // # Allow any user with an account on this server to join this team
+            cy.apiPatchTeam(openTeam.id, {
+                allow_open_invite: true,
+            });
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserInOpenTeam = user;
+                cy.apiAddUserToTeam(openTeam.id, testUserInOpenTeam.id);
+
+                // # Login as test user
+                cy.apiLogin(testUserInOpenTeam);
+            });
+        });
+    });
+
+    it('MM-T452 User with no teams should be able to join an open team from a link in direct messages', () => {
+        // # View 'off-topic' channel
+        cy.visit(`/${openTeam.name}/channels/off-topic`);
+
+        // * Expect channel title to match title passed in argument
+        cy.get('#channelHeaderTitle').
+            should('be.visible').
+            and('contain.text', 'Off-Topic');
+
+        // # Copy full url to channel
+        cy.url().then((openTeamChannelUrl) => {
+            // # From the 'Direct Messages' menu, send the URL of the public channel to the user outside the team
+            sendDirectMessageToUser(testUserOutsideOpenTeam, openTeamChannelUrl);
+
+            // # Logout as test user
+            cy.apiLogout();
+
+            // # Login as user outside team
+            cy.apiLogin(testUserOutsideOpenTeam);
+
+            // # Open direct message from the user in the open team (testUserInOpenTeam)
+            cy.get(`a[href="/${secondTestTeam.name}/messages/@${testUserInOpenTeam.username}"]`).
```

Comment:
We don't normally target an element using this approach but I'm fine as is since it also validates whether the test user is currently viewing the `secondTestTeam`.

Reply:
I understand. In that scenario my alternative was to fetch the unique ID of the channel via API calls like I previously did here: https://github.com/mattermost/mattermost-webapp/blob/a7e0ce41af85b67429ad53e3d7acbb17ef9e627d/e2e/cypress/integration/multi_team_and_dm/close_current_dm_redirects_spec.js#L135

This method requires sysadmin token, which means I had to login as the sysadmin to make the request and that seemed complicated to do in the middle of a test. I wish there was a way to make API calls and supply the token instead of having to rely on the current user session's authentication to do it.
I will keep as is but if I stumble upon a similar scenario in the future I will try to find a better way to do this.

Reply:
Another alternative is to just visit the dm page directly,
```
cy.visit(`/${secondTestTeam.name}/messages/@${testUserInOpenTeam.username}`);
```

Reply:
Yeah Sweet ! There is always a simpler solution :D